src/filters/H5Zzstd_LICENSE.txt

```diff
@@ -0,0 +1,201 @@
+      PARTICULAR PURPOSE. You are solely responsible for determining the
+      appropriateness of using or redistributing the Work and assume any
+      risks associated with Your exercise of permissions under this License.
+
+   8. Limitation of Liability. In no event and under no legal theory,
+      whether in tort (including negligence), contract, or otherwise,
+      unless required by applicable law (such as deliberate and grossly
+      negligent acts) or agreed to in writing, shall any Contributor be
+      liable to You for damages, including any direct, indirect, special,
+      incidental, or consequential damages of any character arising as a
+      result of this License or out of the use or inability to use the
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
```

Comment:
Usually you can just include this snippet, as it states how to obtain the full license

Reply:
I would rather just distribute the entire license text unless there is a significant issue with that. The license pretty clearly says that we must give recipients of the Work or Derivative work the license. It's really not clear to me that a link is sufficient.

https://opensource.stackexchange.com/questions/5320/apache-v2-how-to-properly-fulfill-the-license-in-a-product

>    4. Redistribution. You may reproduce and distribute copies of the
      Work or Derivative Works thereof in any medium, with or without
      modifications, and in Source or Object form, provided that You
      meet the following conditions:
>
>     (a) You must give any other recipients of the Work or
>          Derivative Works a copy of this License; and

Reply:
I agree with @simonbyrne, I also think this folder is not the right place to put the licenses. 

I would follow the precedent base Julia follows when it comes to licensed ported over code.

Reply:
What exactly do you want to do then?

https://www.apache.org/foundation/license-faq.html#Apply-My-Software

> Include a copy of the Apache License, typically in a file called LICENSE, in your work, and consider also including a NOTICE file.

If I look in my Julia source tree, I find the full Apache License in `src/julia/usr/share/licenses/MbedTLS/LICENSE`.

Reply:
There's another copy in `julia/usr/share/licenses/libLLVM/LICENSE.TXT`

Reply:
Perhaps we should just move the licenses to a separate folder different from `src`, e.g. `licenses`.  Or add a `NOTICES.md` file with links to the location of the full license (see e.g. https://tldrlegal.com/license/apache-license-2.0-(apache-2.0) regarding the NOTICE file) 


Alternately,  according to the license only the following boilerplate code is required, but I'm not sure if that applies ifor derivative work.  If we could do  this,  then we could just include the  notice on  files you've ported over. Again, I'm not sure we that's applicable in our case. 

```
   Copyright [yyyy] [author]

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
```





For example, here's what I did when I worked on the `exp` function: https://github.com/JuliaLang/julia/pull/19831/files

I looked through the Julia repo, it does look like there are cases where things are not followed correctly, e.g.: https://github.com/JuliaLang/julia/blob/1389c2fc4af952f5c8b9759cf6fe633995b523f9/base/special/log.jl#L427,
which should include the original licenses.  It also looks like the new ported over code isn't attributing derivative work e.g. https://github.com/JuliaLang/julia/blob/1389c2fc4af952f5c8b9759cf6fe633995b523f9/base/special/log.jl#L427  should include `SLEEF.jl`, which is irrelevant to the current discussion ,but something I noticed.

Reply:
I think the boilerplate is for applying Apache License to your code. That alone, however, is not sufficient to _comply_ with the license when redistributing the work.

https://www.apache.org/licenses/LICENSE-2.0#apply

![image](https://user-images.githubusercontent.com/8062771/141858152-db88c33e-5a6f-4512-915e-0ac60a45e81f.png)

In this case, we are merely redistributing a derived work. We can license our work under whatever license we like and do not need to apply the Apache license to our own work. I assume we want to apply a MIT license to our own work.

If the original work was under a 2-clause BSD license, then the notice that you provided is sufficient. A 2-clause BSD license only requires that you provide the original copyright license. However, the Apache license explicitly requires that we provide a copy of the license.

Reply:
Makes sense, that's what I suspected. In any case, I think we should move them outside the `src` file since that's confusing. Would it not be more appropriate to put them in a root `licenses` folder?

Reply:
I moved the licenses to a root `licenses` folder.